components/Form.js

```diff
@@ -0,0 +1,229 @@
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import { FaTimes } from "react-icons/fa";
+import { mutate } from "swr";
+
+export default function Form({ onClose, onSubmit }) {
+  const [validationError, setValidationError] = useState("");
+
+  const [formData, setFormData] = useState({
+    name: "",
+    categories: "",
+    minAge: "",
+    description: "",
+    minPlayers: "",
+    maxPlayers: "",
+    image: "",
+    yearpublished: "",
+    playtime: "",
+    userCreated: true,
+  });
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setFormData((prevFormData) => ({ ...prevFormData, [name]: value }));
+  };
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    if (
+      !formData.name ||
+      !formData.image ||
+      !formData.categories ||
+      !formData.description ||
+      !formData.yearpublished ||
+      !formData.playtime ||
+      !formData.minAge ||
+      !formData.minPlayers ||
+      !formData.maxPlayers
```

Comment:
If you don't plan to display particularly custom error messages, prefer `required` on the fields